sdk/communication/Azure.Communication.Calling.Server/src/Models/Events/CallRecordingStateChangeEvent.Serialization.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable disable
+
+using System;
+using System.Text.Json;
+using Azure.Core;
+
+namespace Azure.Communication.Calling.Server
+{
+    public partial class CallRecordingStateChangeEvent
+    {
+        /// <summary>
+        /// Deserialize <see cref="CallRecordingStateChangeEvent"/> event.
+        /// </summary>
+        /// <param name="content">The json content.</param>
+        /// <returns>The new <see cref="CallRecordingStateChangeEvent"/> object.</returns>
+        internal static CallRecordingStateChangeEvent Deserialize(string content)
+        {
+            using var document = JsonDocument.Parse(content);
+            JsonElement element = document.RootElement;
+
+            Optional<string> recordingId = default;
+            Optional<CallRecordingState> state = default;
+            Optional<DateTimeOffset> startDateTime = default;
+            Optional<string> conversationId = default;
+            foreach (var property in element.EnumerateObject())
+            {
+                if (property.NameEquals("recordingId") || property.NameEquals("RecordingId"))
```

Comment:
Questions about these Serialization classes v introduced: 
1. Looks like this is all manually written? Why is that?
2. Why do the properties check names with different cases? (e.g recordingId and RecordingId)

Reply:
I created an issue thread in the channel
@navali-msft 

https://teams.microsoft.com/l/message/19:f17d9cf475e74e1a94033cef0d612a67@thread.skype/1622219271717?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e9c1fc3-39df-4486-a26a-456d80e80f82&parentMessageId=1622219271717&teamName=Azure%20Communication%20Services&channelName=SDK&createdTime=1622219271717

Reply:
@minnieliu  - We have added these classes to deserialize the events which user received on the callback. 

The reason for providing deserialization on our end is because of issue with deserializing CommunicationIdentifierModel. As Service side only understands model defined in Communication.Common, so it is sending that information to user and user don't have way to deserialize it.

We will fix point#2.

Reply:
We have serializer and deserialzer for `CommunicationIdentifierModel` in the Shared folder. Please refer to how the chat project is using them.

Reply:
This Deserialization methods are generated if any of the response contains CommunicationIdentifierModel. In our case we don't have any such response. This is done for deserializing Events.

Reply:
This is fixed.